### PR TITLE
test: migrate button tests to rtl

### PR DESCRIPTION
**Pull request by eszthoff**
Story: https://textkernel.atlassian.net/browse/ONEUI-355

### CODE REVIEW

File: src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
+        const view = render(
             <ButtonGroup>
                 <Button key="a">A button</Button>
                 <Button key="b">Another button</Button>
             </ButtonGroup>
         );
-        expect(toJson(wrapper)).toMatchSnapshot();
+        expect(view.asFragment()).toMatchSnapshot();
+        const button = screen.getAllByRole('button');
+        expect(button[0]).toHaveClass('ButtonGroup__button--first');
```
Comment (PavelKirvelTextkernel) [inline src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx:19]: I think we can apply code style add one empty line.
// Act
// Empty line
// Assert

 expect(view.asFragment()).toMatchSnapshot();
const button = screen.getAllByRole('button');
// empty line
expect(button[0]).toHaveClass('ButtonGroup__button--first');
...

@eszthoff What do you think ?

**Review by PavelKirvelTextkernel — CHANGES_REQUESTED** [review]

File: src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx (hunk 1 above)
Comment (eszthoff) [reply src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx:19]: yes, that's a good point. I will change that

File: src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx [hunk 2]
```diff
@@ -1,33 +1,44 @@
...
     it('should render default button correctly', () => {
-        const wrapper = mount(
+        const view = render(
             <ButtonGroup>
                 <Button key="a">A button</Button>
                 <Button key="b">Another button</Button>
             </ButtonGroup>
         );
-        expect(toJson(wrapper)).toMatchSnapshot();
+        expect(view.asFragment()).toMatchSnapshot();
```
Comment (andrentk) [inline src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx:17]: I think it is more clean and htmlish if we use `view.container` which returns an HTMLElement to check in the snapshot. What do you think?

Comment (eszthoff) [reply src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx:17]: I don't have a strong preference, and I haven't seen anything in the best practices on this. I agree with you, that it seems more natural to call `view.container` the `view.asFragment()`.  @PavelKirvelTextkernel  what do you think?

Comment (PavelKirvelTextkernel) [reply src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx:18]: @eszthoff @andrentk  I think yes we can use it

**Review by PavelKirvelTextkernel — APPROVED** [review]

File: src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx (hunk 2 above)
Comment (eszthoff) [reply src/components/ButtonGroup/__tests__/ButtonGroup.spec.tsx:18]: Ok, thanks. I will make a separate PR to change this in the docs and already existing tests.

**Pull request merged**